v8go.cc

```diff
@@ -1297,6 +1297,38 @@ void ObjectSet(ValuePtr ptr, const char* key, ValuePtr prop_val) {
   obj->Set(local_ctx, key_val, prop_val->ptr.Get(iso)).Check();
 }
 
+RtnStrings ObjectGetPropertyNames(ValuePtr ptr) {
+  LOCAL_OBJECT(ptr);
+  RtnStrings rtn = {};
+
+  Local<Array> names;
+  if (!obj->GetPropertyNames(local_ctx).ToLocal(&names)) {
+    rtn.error = ExceptionError(try_catch, iso, local_ctx);
+    return rtn;
+  }
+
+  uint32_t length = names->Length();
+  const char** strings = (const char**)malloc(length * sizeof(const char*));
+
+  for (uint32_t i = 0; i < length; i++) {
```

Comment:
Question for anyone but when should we be using `uint32_t` vs `int`? [Other for-loops](https://github.com/rogchap/v8go/blob/5e91d3d9dcabd2986f901b6b31590e49fc3c4dd8/v8go.cc#L502) in here use `int`

Reply:
We should try to match the loop bounds to avoid integer overflows.

Using a `uint32_t` makes sense here since [v8::Array::Length()](https://v8.github.io/api/head/classv8_1_1Array.html#ad6b33c7dd543044ca2311c000a1642dd) returns a uint32_t.

The iteration count for the loop you link to uses
[v8::FunctionCallbackInfo::Length()](https://v8.github.io/api/head/classv8_1_1FunctionCallbackInfo.html#af97dd3f1cb01ed039f9479152ad63a84) which returns an `int`, so using an `int` makes the most sense there.